docs/deployment/pull-backup.rst

```diff
@@ -174,3 +188,116 @@ directly extract it without the need of mounting with SSHFS:
 Note that in this scenario the tar format is the limiting factor – it cannot
 restore all the advanced features that BorgBackup supports. See
 :ref:`borg_export-tar` for limitations.
+
+socat
+=====
+
+In this setup a SSH connection from the backup server to the client is
+established that uses SSH reverse port forwarding to transparently
+tunnel data between UNIX domain sockets on the client and server and the socat
+tool to connect these with the borg client and server processes, respectively.
+
+The program socat has to be available on the backup server and on the client
+to be backed up.
+
+When **pushing** a backup the borg client (holding the data to be backed up)
+connects to the backup server via ssh, starts ``borg serve`` on the backup
+server and communicates via standard input and output (transported via SSH)
+with the process on the backup server.
+
+With the help of socat this process can be reversed. The backup server will
+create a connection to the client (holding the data to be backed up) and will
+**pull** the data.
+
+In the following example *borg-server* connects to *borg-client* to pull a backup.
+
+To provide a secure setup sockets should be stored in ``/run/borg``, only
+accessible to the users that run the backup process. So on both systems,
+*borg-server* and *borg-client* the folder ``/run/borg`` has to be created::
+
+   sudo mkdir -m 0700 /run/borg
+
+On *borg-server* the socket file is opened by the user running the ``borg
+serve`` process writing to the repository
+so the user has to have read and write permissions on ``/run/borg``::
+
+   borg-server:~$ sudo chown borgs /run/borg
+
+On *borg-client* the socket file is created by ssh, so the user used to connect
+to *borg-client* has to have read and write permissions on ``/run/borg``::
+
+   borg-client:~$ sudo chown borgc /run/borg
+
+On *borg-server*, we have to start the command ``borg serve`` and make its
+standard input and output available to a unix socket::
+
+   borg-server:~$ socat UNIX-LISTEN:/run/borg/reponame.sock,fork EXEC:"borg serve --append-only --restrict-to-path /path/to/repo"
+
+.. note::
+   When used in production you may also use systemd socket-based activation
+   instead of socat on the server side. You would wrap the ``borg serve`` command
+   in a `service unit`_ and configure a matching `socket unit`_
+   to start the service whenever a client connects to the socket.
+
+   .. _service unit: https://www.freedesktop.org/software/systemd/man/systemd.service.html
+   .. _socket unit: https://www.freedesktop.org/software/systemd/man/systemd.socket.html
+
+Socat will wait until a connection is opened. Then socat will execute the
+command given, redirecting Standard Input and Output to the unix socket. The
+optional arguments for ``borg serve`` are not necessary but a sane default.
```

Comment:
I'd move that paragrah before the systemd note, right after the socat command that it explains.

Reply:
I've added 511460a7